hopr/hopr-lib/src/lib.rs

```diff
@@ -259,7 +261,8 @@ where
                     }
                 }
                 ChainEventType::NetworkRegistryUpdate(address, allowed) => {
-                    match db.read().await.get_packet_key(&address).await {
```

Comment:
`match` and `if let` statements are the most dangerous ones, as the lock is being held across the entire block, not just the statement.

Reply:
![Been there... ](https://media1.tenor.com/m/0FJbp1RGsF0AAAAC/elrond-lotr.gif)